inc/aiosp_common.php

```diff
@@ -182,4 +189,256 @@ public static function make_xml_safe( $tag, $value ) {
+	 */
+	public static function attachment_url_to_postid_query_2() {
+		global $wpdb;
+
+		$tmp_arr = array();
+		$results_2 = $wpdb->get_results(
+			"SELECT post_id, meta_value FROM {$wpdb->postmeta} WHERE `meta_key` = '_wp_attachment_metadata' AND `meta_value` != '" . serialize( array() ) . "';",
+			ARRAY_A
+		);
+		if ( $results_2 ) {
+			for ( $i = 0; $i < count( $results_2 ); $i++ ) {
+				// TODO Investigate potentual memory leak(s); currently with unserialize.
+				$meta_value = unserialize( $results_2[ $i ]['meta_value'] );
+
+				// TODO Needs Discussion: Should this be added? To handle errors better instead of suspecting aioseop is at fault and lessen support threads.
+				/**
+				 * This currently handles "warning" notices with unserialize which normally can't be handled with a try/catch.
+				 * However, this notice should be identified and corrected; which is seperate from the plugin, but
+				 * can also triggered by the plugin.
+				 *
+				 * @see aiosp_common::error_handle_images()
+				 * @see set_error_handler()
+				 * @link http://php.net/manual/en/function.set-error-handler.php
+				 * @see restore_error_handler()
+				 * @link http://php.net/manual/en/function.restore-error-handler.php
+				 */
+				/*
+				set_error_handler( 'aiosp_common::error_handle_images' );
+				try {
+					$meta_value = unserialize( $results_2[ $i ]['meta_value'] );
+				} catch ( Exception $e ) {
+					unset( $meta_value );
+					restore_error_handler();
+					continue;
+				}
+				restore_error_handler();
+				*/
+
+				// Images and Videos use different variable structures.
+				if ( false === $meta_value || ! isset( $meta_value['file'] ) && ! isset( $meta_value['sizes'] ) ) {
+					continue;
+				}
+
+				// Set the URL => PostIDs.
+				$uploads_dir = wp_upload_dir();
+				$custom_img_base_url = $uploads_dir['baseurl'] . '/' . str_replace( basename( $meta_value['file'] ), '', $meta_value['file'] );
+				foreach ( $meta_value['sizes'] as $image_size_arr ) {
+					$tmp_arr[ md5( ( $custom_img_base_url . $image_size_arr['file'] ) ) ] = $results_2[ $i ]['post_id'];
+				}
+
+				unset( $meta_value );
+			}
+		}
+
+		$results_2 = $tmp_arr;
+		unset( $tmp_arr );
+
+		return $results_2;
+	}
+
+	/**
+	 * Error Hand Images
+	 *
+	 * Unused/Conceptual function potentually used in `aiosp_common::attachment_url_to_post_id_query_2()`.
+	 * This is to handle errors where a normal try/catch wouldn't have the exception needed to catch.
+	 *
+	 * @see aiosp_common::attachment_url_to_post_id_query_2()
+	 *
+	 * @param $errno
+	 * @param $errstr
+	 * @param $errfile
+	 * @param $errline
+	 * @return bool
+	 * @throws ErrorException
+	 */
+	public static function error_handle_images( $errno, $errstr, $errfile, $errline ) {
+		// Possibly handle known issues differently.
+		// Handles unserialize() warning notice.
+		if ( 8 === $errno || strpos( $errstr , 'unserialize():' ) ) {
+			throw new ErrorException( $errstr, $errno, 0, $errfile, $errline );
```

Comment:
ErrorException might not work in PHP versions below 5.1.0

Reply:
I believe that is true. However, this was mainly used for dev purposes.

This is also in a disabled method for conceptual use.

Reply:
maybe just change ErrorException into Exception if you are going to retain the method for future use?

Reply:
I believe that type of function was required for that type of warning. Because I remember trying Exception and the warning wasn't being caught.

Right now it's all disabled, and would rather leave it as what I know was last functional.